SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SchemaGenObjectStub.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "CoreMinimal.h"
+
+#include "SchemaGenObjectStub.generated.h"
+
+UCLASS()
+class USchemaGenObjectStub : public UObject
+{
+	GENERATED_BODY()
+public:
+
+	UPROPERTY(Replicated)
+	int IntValue;
+
+	UPROPERTY(Replicated)
+	bool BoolValue;
+};
+
+UCLASS(SpatialType)
+class USpatialTypeObjectStub : public UObject
+{
+	GENERATED_BODY()
+};
+
+UCLASS()
+class UChildOfSpatialTypeObjectStub : public USpatialTypeObjectStub
+{
+	GENERATED_BODY()
+};
+
+UCLASS(NotSpatialType)
+class UNotSpatialTypeObjectStub : public UObject
+{
+	GENERATED_BODY()
+};
+
+UCLASS()
+class UChildOfNotSpatialTypeObjectStub : public UNotSpatialTypeObjectStub
+{
+	GENERATED_BODY()
+};
+
+UCLASS()
+class UNoSpatialFlagsObjectStub : public UObject
+{
+	GENERATED_BODY()
+};
+
+UCLASS()
+class UChildOfNoSpatialFlagsObjectStub : public UNoSpatialFlagsObjectStub
+{
+	GENERATED_BODY()
+};
+
+UCLASS(SpatialType)
+class ASpatialTypeActor : public AActor
+{
+	GENERATED_BODY()
+};
+
+UCLASS()
+class ANonSpatialTypeActor : public AActor
+{
+	GENERATED_BODY()
+};
+
+UCLASS(SpatialType)
+class USpatialTypeActorComponent : public UActorComponent
+{
+	GENERATED_BODY()
+};
+
+UCLASS(SpatialType)
+class ASpatialTypeActorWithActorComponent : public AActor
+{
+	GENERATED_UCLASS_BODY()
+
+	UPROPERTY(Replicated)
+	USpatialTypeActorComponent* SpatialActorComponent;
+};
+
+UCLASS(SpatialType)
+class ASpatialTypeActorWithMultipleActorComponents : public AActor
+{
+	GENERATED_UCLASS_BODY()
+
+	UPROPERTY(Replicated)
+	USpatialTypeActorComponent* FirstSpatialActorComponent;
+
+	UPROPERTY(Replicated)
+	USpatialTypeActorComponent* SecondSpatialActorComponent;
+};
+
+UCLASS(SpatialType)
+class ASpatialTypeActorWithMultipleObjectComponents : public AActor
```

Comment:
Should probably also have a test for `SpatialType` classes which contain non-replicated components and make sure we don't generate schema for them.

Reply:
+1
Also custom replicated subobjects (non-UActorComponents) but I think they can wait for now, add them into the follow up ticket, otherwise we'll never finish this.